swiftide/src/transformers/file_to_context_tree_sitter.rs

```diff
@@ -0,0 +1,60 @@
+//! Chunk code using tree-sitter
+use anyhow::Result;
+use async_trait::async_trait;
+use derive_builder::Builder;
+
+use crate::{
+    indexing::Node,
+    integrations::treesitter::{CodeSummarizer, SupportedLanguages},
+    Transformer,
+};
+
+pub const NAME: &str = "Context (code)";
+
+#[derive(Debug, Clone, Builder)]
+#[builder(pattern = "owned", setter(into, strip_option))]
+pub struct FileToContextTreeSitter {
+    summarizer: CodeSummarizer,
+}
+
+impl FileToContextTreeSitter {
```

Comment:
FileToContextTreesiter gives me little hints on what the intended use is, and I'm still unsure what summarizer will return.